pkg/specgenutil/specgen.go

```diff
@@ -324,6 +325,15 @@ func FillOutSpecGen(s *specgen.SpecGenerator, c *entities.ContainerCreateOptions
 		return err
 	}
 
+	// Set the image OS - if the option is set, use that, otherwise default
+	// to the host OS.  This is needed on FreeBSD which supports both native
+	// and emulated Linux containers.
+	if c.OS != "" {
+		s.ImageOS = c.OS
+	} else {
+		s.ImageOS = runtime.GOOS
+	}
```

Comment:
This will not work, we cannot ever propagate this on the client side.
If you run the remote client on mac or windows all images will break.
This once again shows how desperately we need actual windows/macOS testing in CI not just build tests.

Reply:
Nice catch!  So we'd need to make this behavior conditional on FreeBSD-only.

> This once again shows how desperately we need actual windows/macOS testing in CI not just build tests.

Absolutely

Reply:
No not even bsd only, we just merged podman service support for bsd which means the remote client could connect from a different host os there as well.

This code must be moved somewhere in the code path that is only called on the server side, unfortunately our code structure in pkg/specgen does not make it clear what is called where.

Reply:
Third attempt - this iteration queries the image itself for the OS. This seems better because it just goes with whatever image upper-level policy ended up fetching.